rdflib/tools/chunk_serializer.py

```diff
@@ -0,0 +1,136 @@
+"""
+This file provides a single function `serialize_in_chunks()` which can serialize a
+Graph into a number of NT files with a maximum number of triples or maximum file size.
+
+There is an option to preserve any prefixes declared for the original graph in the first
+file, which will be a Turtle file.
+"""
+
+import os
+from pathlib import Path
+
+try:
+    from rdflib import Graph, Literal
+except Exception:
+    import sys
+
+    sys.path.insert(0, str(Path(__file__).parent.parent.parent.absolute()))
+    from rdflib import Graph, Literal
+
+__all__ = ["serialize_in_chunks"]
+
+
+def serialize_in_chunks(
+    g: Graph,
+    max_triples: int = 10000,
+    max_file_size_kb: int = None,
+    file_name_stem: str = "chunk",
+    output_dir: Path = Path(__file__).parent,
+    first_file_contains_prefixes: bool = False,
+):
+    """
+    Serializes a given Graph into a series of n-triples with a given length
+
+    max_file_size_kb:
```

Comment:
One challenge with using kB in the naming is that it is a bit ambiguous in whether this is 1000 bytes or 1024 bytes. It may be slightly less ambiguous to just use size in bytes, but bytes would be too small for most cases.

Reply:
I know what you mean: I wrestled with B, kB, MB etc. I think kB is "ok", as in there are no perfect solutions and it's not likely critical what the value is, e.g. 1000 or 1024, so long as it's approximate.